src/java/org/apache/cassandra/net/RequestCallbacks.java

```diff
@@ -130,10 +130,11 @@ public void addWithExpiration(AbstractWriteResponseHandler<?> cb,
         assert previous == null : format("Callback already exists for id %d/%s! (%s)", message.id(), to.endpoint(), previous);
     }
 
+    @Nullable
     <In,Out> IVersionedAsymmetricSerializer<In, Out> responseSerializer(long id, InetAddressAndPort peer)
     {
         CallbackInfo info = get(id, peer);
-        return info == null ? null : info.responseVerb.serializer();
+        return info == null || info.responseVerb == null ? null : info.responseVerb.serializer();
```

Comment:
I think it would be good to have a comment here to explain why this is being done.